tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -126,134 +132,305 @@ void validate(
 }
 } // namespace
 
-TEST_P(DistributedTransformerTest, MLP_Layer) {
-  auto [use_aten_matmul, dtype] = GetParam();
-  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
-  FusionGuard fg(fusion.get());
-  auto mesh = DeviceMesh::createForNumDevices(communicator_->size());
-  at::ScalarType at_dtype = data_type_to_aten(dtype);
+std::vector<at::Tensor> reference_mlp(
+    at::Tensor x,
+    at::Tensor w0,
+    at::Tensor b0,
+    at::Tensor w1,
+    at::Tensor b1,
+    at::ScalarType at_dtype) {
+  at::manual_seed(0);
```

Comment:
Is https://github.com/NVIDIA/Fuser/blob/88f6a6295fc652f10b4c86416744c6b69eb275cc/tests/cpp/utils.h#L500 not good enough?

Reply:
Sorry I wasn't clear. Normally, I'd expect the random seed to be set only once and at the beginning of the program, so https://github.com/NVIDIA/Fuser/blob/88f6a6295fc652f10b4c86416744c6b69eb275cc/tests/cpp/utils.h#L500 makes sense. I've yet to understand why we need to call `at::manual_seed` at all in individual tests.

Reply:
Ahh, it's to ensure the same seed is used for the reference and nvfuser program for dropout.

Reply:
Got it!

Solution-wise, there's the seed and there's the state. The common pattern I've seen is to `torch.random.fork_rng` the reference implementation, e.g., https://github.com/pytorch/pytorch/blob/53a5e0f1a8ef5087eeb57deb46f4b0c488fdbcc8/torch/jit/_trace.py#L282-L284. This way, the ref and the actual start with the same state without reseting the seed, which is supposed to be set once. Doing this in C++ may be more cumbersome. I haven't tried this, but I saw get_state and set_state get exposed
https://github.com/pytorch/pytorch/blob/53a5e0f1a8ef5087eeb57deb46f4b0c488fdbcc8/aten/src/ATen/cuda/CUDAGeneratorImpl.cpp#L323 and can be accessed via https://github.com/pytorch/pytorch/blob/53a5e0f1a8ef5087eeb57deb46f4b0c488fdbcc8/torch/csrc/api/src/cuda.cpp#L32.

Reply:
Dropout exposes the philox seed and the tests do set it (https://github.com/NVIDIA/Fuser/blob/main/tests/cpp/test_multidevice_transformer.cpp#L92)....so this test might be ok without reseting the seed. 
I will experiment!

Reply:
> I will experiment!

Thanks! 

https://github.com/NVIDIA/Fuser/blob/main/tests/cpp/test_multidevice_transformer.cpp#L92-L94 looks suspicious. Instead of passing in `seed=0,offset=0`, are we supposed to pass in `Generator::seed()` and `Generator::get_offset()`?

Reply:
Yes, I think that is correct. Looks like the defualt is seed is 0 which is why the code ran correctly.